numpyro/primitives.py

```diff
@@ -232,7 +233,17 @@ def process_message(self, msg):
         cond_indep_stack = msg['cond_indep_stack']
         frame = CondIndepStackFrame(self.name, self.dim, self.subsample_size)
         cond_indep_stack.append(frame)
-        batch_shape = self._get_batch_shape(cond_indep_stack)
+        expected_shape = self._get_batch_shape(cond_indep_stack)
+        dist_batch_shape = msg['fn'].batch_shape if msg['type'] == 'sample' else ()
+        overlap_idx = len(expected_shape) - len(dist_batch_shape)
+        if overlap_idx < 0:
+            raise ValueError('Expected dimensions within plate = {}, which is less than the '
+                             'distribution\'s batch shape = {}.'.format(len(expected_shape), len(dist_batch_shape)))
+        trailing_shape = expected_shape[overlap_idx:]
+        if sum(map(sub, lax.broadcast_shapes(trailing_shape, dist_batch_shape), dist_batch_shape)) > 0:
+            raise ValueError('Distribution batch shape = {} cannot be broadcast to {}.'
+                             .format(trailing_shape, dist_batch_shape))
```

Comment:
nit: I guess you mean `"Distribution batch shape = {} cannot be broadcasted from the trailing plate shape {}.".format(dist_batch_shape, trailing_shape)`?

Reply:
I meant to rather than from, since this should fail precisely when we try to broadcast up an existing distribution batch dim. Let me add the detail about the trailing plate dim.

Reply:
Sorry, arguments here are reversed, will fix.